build/__main__.py

```diff
@@ -27,15 +28,34 @@ def _error(msg, code=1):  # type: (str, int) -> None  # pragma: no cover
     exit(code)
 
 
-def build(srcdir, outdir, distributions, config_settings=None, skip_dependencies=False):
-    # type: (str, str, List[str], Optional[ConfigSettings], bool) -> None
+def _build_in_isolated_env(builder, outdir, distributions):  # type: (ProjectBuilder, str, List[str]) -> None
+    with IsolatedEnvironment() as env:
+        env.install(builder.build_dependencies)
+        for distribution in distributions:
+            builder.build(distribution, outdir)
+
+
+def _build_in_current_env(builder, outdir, distributions, skip_dependencies=False):
+    # type: (ProjectBuilder, str, List[str], bool) -> None
+    for dist in distributions:
+        if not skip_dependencies:
+            missing = builder.check_dependencies(dist)
+            if missing:
+                _error('Missing dependencies:' + ''.join(['\n\t' + dep for dep in missing]))
```

Comment:
instead of addition consider using ``'{}{}'.format``? ( I know I don't like + for string addition, makes it hard to read and type inspect)

Reply:
This should come in a separate PR, I just move the code to a different place.